assets/src/components/parts/janus-input-number/InputNumber.tsx

```diff
@@ -120,81 +198,35 @@ const InputNumber: React.FC<PartComponentProps<InputNumberModel>> = (props) => {
               if (sValue !== undefined) {
                 setInputNumberValue(sValue);
               }
-              const sCssClass = initStateFacts[`stage.${id}.customCssClass`];
-              if (sCssClass !== undefined) {
-                setCssClass(sCssClass);
-              }
+
               if (payload.mode === contexts.REVIEW) {
                 setEnabled(false);
               }
             }
             break;
         }
       };
-      const unsub = subscribeToNotification(props.notify, notificationType, handler);
+      const unsub = subscribeToNotification(notify, notificationType, handler);
       return unsub;
     });
     return () => {
       notifications.forEach((unsub) => {
         unsub();
       });
     };
-  }, [props.notify]);
-
-  useEffect(() => {
-    let pModel;
-    let pState;
-    if (typeof props?.model === 'string') {
-      try {
-        pModel = JSON.parse(props.model);
-        setModel(pModel);
-      } catch (err) {
-        // bad json, what do?
-      }
-    }
-    if (typeof props?.state === 'string') {
-      try {
-        pState = JSON.parse(props.state);
-        setState(pState);
-      } catch (err) {
-        // bad json, what do?
-      }
-    }
-    if (!pModel) {
-      return;
-    }
-    initialize(pModel);
-  }, [props]);
+  }, [id, notify]);
 
   useEffect(() => {
-    if (!ready) {
-      return;
-    }
-    props.onReady({ id, responses: [] });
```

Comment:
Didn't see a reason we couldn't just call onReady after calling setReady(true) above, fewer useEffect calls seems better than more.